diff --git a/pytorch/module/linear.py b/pytorch/module/linear.py
index cf7f589..c608880 100644
--- a/pytorch/module/linear.py
+++ b/pytorch/module/linear.py
@@ -215,6 +215,7 @@ class _Linear(torch.autograd.Function):
                 )
 
         else:  # Do not all-gather input tensor
+            inputmat = cast_if_needed(inp, activation_dtype)  # Cast for AMP
             if fp8 or debug:
                 if isinstance(inputmat, QuantizedTensorBase):
                     inputmat.update_usage(rowwise_usage=True)
@@ -226,8 +227,6 @@ class _Linear(torch.autograd.Function):
                     )
                     inputmat = input_quantizer(inputmat)
                     own_quantized_input = True
-            else:
-                inputmat = cast_if_needed(inp, activation_dtype)  # Cast for AMP
             inputmat_total = inputmat
         nvtx_range_pop(f"{nvtx_label}.input_cast_comm")
         # ------------------------------------------------------
